universal-login-react/src/ui/TopUp/ChooseTopUpMethod.tsx

```diff
@@ -25,11 +25,19 @@ export interface ChooseTopUpMethodProps {
 }
 
 export const ChooseTopUpMethod = ({sdk, contractAddress, onPayClick, topUpClassName, logoColor, isDeployment, onGasParametersChanged}: ChooseTopUpMethodProps) => {
+  const [gasParameters, setGasParameters] = useState<GasParameters | undefined>(undefined);
   if (isDeployment) {
     ensureNotNull(onGasParametersChanged, MissingParameter, 'onGasParametersChanged');
   }
+  const gasParametersChanged = (gasParameters: GasParameters) => {
+    if (isDeployment) {
+      setGasParameters(gasParameters);
+      onGasParametersChanged!(gasParameters);
+    }
+  };
   const [topUpMethod, setTopUpMethod] = useState('');
   const methodSelectedClassName = topUpMethod !== '' ? 'method-selected' : '';
+  const getMinimalAmount = (gasParameters?: GasParameters) => gasParameters && safeMultiply(DEPLOYMENT_REFUND.mul(3).div(2), gasParameters.gasPrice);
```

Comment:
What is this?  😨

Reply:
As @marekkirejczyk said we just show 150% of deployment fee as a minimal amount. So I can rename and move this function